### PR TITLE
Issue 564 - change message ( fixed english only )

### PR DESCRIPTION
**Pull request by aitkeni**
## Description

resolves issue 564. the message was confusing so was changed (in the english resourse only ) to the suggested stopped when location services are disabled or stopped at user request
## How to test
1 go to settings,
click on stop logging text or icon, 
see message appears.  it should now read
Location Tracking was disabled
Stopped at user request.
2 go to settings
turn off location services from OS.
click on stop logging text or icon
see message appears.  it should now read
Location Tracking was disabled
Stopped.

Still todo other languages (please advise!)
(In another locale the english text will appear until the translation is present I think)

### CODE REVIEW

**Comment by tremblerz** [comment]
Can you add more verbose message keys and preferable title also?
Something like - "Device location has been disabled. Please enable it in the settings" and "Location access to Safe Paths has been removed. Please enable it in the settings for effective logging" (Or instead we can do the same on-boarding kind of process also from UX perspective)
Ex. - Uber says: "For a better location experience, turn on the device location"

Given that the PR is dedicated towards only changing the message which is same for the both case right now, I would suggest only change text and get it in, and make a ticket for much in depth UI/UX for the events of location being disabled/enabled etc.

**Comment by aitkeni** [comment]
Makes sense.  Leave it with me!

**Comment by aitkeni** [comment]
Actually One clarification @tremblerz , the proposed change will distinguish between if the location services have been switched off externally or if its the user clicking the stop tracking icon.  So what you are suggesting is to change the message when location services have been disabled externally
to
   "location_disabled_title": "Device location has been disabled. Please enable it in the settings",
   "location_disabled_message": "Location access to Safe Paths has been removed. Please enable it in the settings for effective logging.",

I cant help thinking that the title and message duplicate information, as they are both displayed together at all times.  the message can show more info as the font is smaller.
would it not make more sense to have the title convey the current state, and the message to convey the reason for the state and the suggested resolution. 
"location_disabled_title": "Device location has been disabled.",
   "location_disabled_message": "Location access to Safe Paths has been removed. Please enable it in the settings for effective logging.",

How does that sound?

**Comment by alpita-masurkar** [comment]
> Actually One clarification @tremblerz , the proposed change will distinguish between if the location services have been switched off externally or if its the user clicking the stop tracking icon. So what you are suggesting is to change the message when location services have been disabled externally
> to
> "location_disabled_title": "Device location has been disabled. Please enable it in the settings",
> "location_disabled_message": "Location access to Safe Paths has been removed. Please enable it in the settings for effective logging.",
> 
> I cant help thinking that the title and message duplicate information, as they are both displayed together at all times. the message can show more info as the font is smaller.
> would it not make more sense to have the title convey the current state, and the message to convey the reason for the state and the suggested resolution.
> "location_disabled_title": "Device location has been disabled.",
> "location_disabled_message": "Location access to Safe Paths has been removed. Please enable it in the settings for effective logging.",
> 
> How does that sound?

On that note, you also don't need 'has been' in these two strings. The meaning is still conveyed and is shorter and might look better on screen.

"location_disabled_title": "Device location disabled.",
"location_disabled_message": "Location access to Safe Paths removed. Please enable it in the settings for effective logging.",

**Comment by aitkeni** [comment]
agreed

**Comment by alpita-masurkar** [comment]
#564

**Comment by summetj** [comment]
Sorry, missed this pull request because it was not linked to the issue https://github.com/tripleblindmarket/covid-safe-paths/issues/564.   Should be merged with https://github.com/tripleblindmarket/covid-safe-paths/pull/647

**Comment by tstirrat** [comment]
I did a test on this (Pixel 2 emulator), and it gets stuck showing "stopped at user request" even though I actually have location permissions disabled in the OS.

I would expect it to say "Location access to Safe Paths removed. Please enable it in the settings for effective logging." right?

I suspect we need more thorough checks..

Can you show a video of how it should be working?

**Comment by aitkeni** [comment]
Thanks for helping me on this @tstirrat . 
heres a vid of the way the new messages are showing.
[2020-04-25 00-53-20.mkv.zip](https://github.com/tripleblindmarket/covid-safe-paths/files/4532044/2020-04-25.00-53-20.mkv.zip)
Unfortunately I just noticed that the new suggested text for when location services are disabled was too wordy and spills over uncomfortably onto two lines and the message is truncated.  
image
I think I should be shortened again.
Also I could reproduce the issue you are seeing when the loc services are disabled then the ui shows active when bringing the app back into the foreground
[2020-04-25 01-10-59.mkv.zip](https://github.com/tripleblindmarket/covid-safe-paths/files/4532059/2020-04-25.01-10-59.mkv.zip)
i dont think this is an issue i have introduced with my change (but i could be wrong) 
If I confirm its an existing defect then I will raise a new issue and I can take a look this weekend, but will shorten that message first.
many thx.

**Comment by aitkeni** [comment]
just confirmed that this defect exists without my changes.  in fact when location services disabled it gets a bit messed up.  the app can think location tracking is enabled when its not as well as getting stuck at Location tracking disabled ( which is the original message before my changes )  Ill raise an issue for this.

**Comment by aitkeni** [comment]
issue #685 raised

**Comment by aitkeni** [comment]
apologies for the delay in updating.  I have shortened the messages so they fit into the android notification message area.  I have also significantly changed the word 'securely' to 'anonymously' in the Enabled message.  I think it is more accurate and better for PR (it seems all the press is talking about how contact tracing is bad for your privacy rights) so I thought this would be a better message and would make our app rise above...  regarding the issues you saw @tstirrat i have a separate PR coming to fix that issue (issue-685)

**Comment by tstirrat** [comment]
Superseded by #725
